src/modules/Content/index.tsx

```diff
@@ -212,31 +216,50 @@ function useSearchState(contentQuery: QueryResult<ContentQuery, ContentQueryVari
      * depending on the search matches
      */
     const body = contentQuery.state.data?.content[0]?.body;
-    const bodyWithMatches: ContentBody[] = useMemo(
-        () =>
-            state.searchTerm && body
-                ? findAll({
-                      autoEscape: true,
-                      caseSensitive: false,
-                      searchWords: [state.searchTerm],
-                      textToHighlight: body,
-                  }).map(({ highlight, start, end }, index) => ({
-                      highlight,
-                      id: `content-body-search-term-chunk-${index}`,
-                      text: body.substr(start, end - start),
-                  }))
-                : [
-                      {
-                          highlight: false,
-                          id: 'content-body',
-                          text: body || '',
-                      },
-                  ],
-        [body, state.searchTerm]
-    );
+    const bodyWithMatches: ContentBody[][] | null = useMemo(() => {
+        if (body) {
+            const nodes: NodeListOf<ChildNode> = new DOMParser().parseFromString(body, 'text/html').body.childNodes;
+            // Each ContentBody array inside this 2D array represents a paragraph
+            const chunks: ContentBody[][] = [];
+            // Map and filter methods are not available for NodeListOf
+            nodes.forEach((node, nodeIndex) => {
+                // We may need to support additional tags besides <p>
```

Comment:
Do you even need to check for `P`? Can you just use `textContent` unconditionally or does that cause problems?